### PR TITLE
Require devs to write release note snippets

### PR DESCRIPTION
**Pull request by d3zd3z**
This PR documents the process whereby devs are required to write release note snippets as part of their changes. These notes will be reviewed together with the code.

I would like to merge this immediately after the 1.10.0 release, and start enforcing the release notes for PRs after this release.

### CODE REVIEW

**Comment by d3zd3z** [comment]
Implements #1606

File: docs/release.md [hunk 1]
```diff
@@ -24,6 +24,12 @@ Before making a release, update the `docs/release-notes.md` file
 to describe the release. This should be a high-level description of
 the changes, not a list of the git commits.
 
+Provided that changes going into the release have followed the
+contributing guidelinkes, this should mostly consist of collecting the
```
Comment (utzig) [inline docs/release.md:28]: Not sure if there is a typo here, should this be "contribution guidelines"?

File: docs/SubmittingPatches.md [hunk 2]
```diff
@@ -30,6 +30,19 @@ Certificate of Origin" located at the bottom of this page:
   see this [link]
   (https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
 
+## Release notes
+
+To facility creating release notes at release time, all non-trivial
+changes must include a release note snippet with the pull request.
+This can be either a separate commit, or as part of a single commit
+(generally, if there are multiple commits to the pull request, the
+release note snippet should be a separate commit, and a pull request
```
Comment (de-nordic) [inline docs/SubmittingPatches.md:39]: I agree on separate commit but separate PR should not be mandatory. I prefer to have release notes with the PR that introduces changes, when possible.

File: docs/release-notes.d/00readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- Documentation improvements.
+- Strictly internal changes to the code that won't be visible to users
+  of the code.
+
+## Release note format
+
+Release notes are included in files under this `docs/release-notes.d`
+directory and have a name of `*.md`.  They will be included in the
+`release-notes.md` file, and should be formatted as a Markdown list
+entry.
```
Comment (de-nordic) [inline docs/release-notes.d/00readme.md:23]: How will they be ordered? Can we establish that files are named by PR number that brings them or brings the change they are for? Otherwise we are in for a name conflict at some point anyway.

File: docs/release-notes.d/00readme.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+Choose a filename that is related to what this change does.  The names
+are not used for anything in particular, but to keep the files
```
Comment (de-nordic) [inline docs/release-notes.d/00readme.md:26]: Can we establish that files are named by PR number that brings them or brings the change they are for? Otherwise we are in for a name conflict at some point anyway.

File: docs/release-notes.d/00readme.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+Generally a release note entry should be created for changes that:
+
+- Fix bugs in the code.
+- Implement new features.
+- Change existing behavior.
+
+Release notes are generally not needed for:
+
+- Documentation improvements.
```
Comment (de-nordic) [inline docs/release-notes.d/00readme.md:14]: Sometimes documentation improvement may require release note, for example if documentation states some MTU and it is incorrect, by mistake, and code has the right one, the documentation will be corrected which requires external code to follow.

File: docs/release-notes.d/00readme.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Pending release notes directory
+
+This directory contains release note entries that have not been merged
+into the main release-notes.md document.
+
+Generally a release note entry should be created for changes that:
```
Comment (de-nordic) [inline docs/release-notes.d/00readme.md:6]: Should we have some github label that will tell that change will require or is providing release note?

File: docs/SubmittingPatches.md (hunk 2 above)
Comment (d3zd3z) [reply docs/SubmittingPatches.md:39]: I can try to clarify this, because I agree the notes should definitely be part of the same PR.

File: docs/release-notes.d/00readme.md (hunk 3 above)
Comment (d3zd3z) [reply docs/release-notes.d/00readme.md:23]: Unsure. Normally, they are ordered by what order they get committed in. I suspect I might be able to come up with some git/bash magic that would combine them in commit order, probably put that in etc to help

File: docs/release-notes.d/00readme.md (hunk 4 above)
Comment (d3zd3z) [reply docs/release-notes.d/00readme.md:26]: It hasn't really been an issue with mbed TLS. Plus, you don't really know the pr until you push it.

File: docs/release-notes.d/00readme.md (hunk 6 above)
Comment (d3zd3z) [reply docs/release-notes.d/00readme.md:6]: Maybe a CI job that marks changes based on some rules, and then we can override that if it gets it wrong.

File: docs/release-notes.d/00readme.md (hunk 4 above)
Comment (de-nordic) [reply docs/release-notes.d/00readme.md:26]: People can always amend commits once PR has been created...

**Comment by d3zd3z** [comment]
Once reviewers agree, I think this is ready to merge for the next release. @de-nordic, @utzig

**Review by de-nordic — APPROVED** [review]
Good enough for me. Thanks!
